### PR TITLE
Zipcode fix

### PR DESCRIPTION
**Pull request by aesharpe**
This ensures the zipcode fields in EIA are all named either `zip_code`, `zip_code_4` or `XX_zip_code` depending on their tabular context. It also establishes a `zero_pad_zips` function that retains preceding zeros in a zipcode and turns 0 zipcodes to `nan`. This function is written in the `pudl.helper` module and called in the `convert_cols_dtype` function. It does not yet translate to the output tables.

### CODE REVIEW

File: src/pudl/helpers.py [hunk 1]
```diff
@@ -1085,12 +1099,16 @@ def zero_pad_zips(zip_series, n_digits):
...
+        return series.str.replace(r'[\.]+\d*', '', regex=True)
+
     zip_series = (
-        pd.to_numeric(zip_series)  # Make sure it's all numerical values
-        .astype(pd.Int64Dtype())  # Make it a (nullable) Integer
-        .fillna(0)  # fill the NA
-        .astype(str).str.zfill(n_digits)  # Make it a string and zero pad it.
-        .astype(pd.StringDtype())  # Make it nullable
+        zip_series
+        .astype(pd.StringDtype())
```
Comment (cmgosnell) [inline src/pudl/helpers.py:1107]: did converting these to nullable ints not work here? is this because of the canadian zips?

**Review by cmgosnell — APPROVED** [review]

**Review by cmgosnell — COMMENTED** [review]
This all looks good to me, but do you know why the tests are failing here?

File: src/pudl/helpers.py (hunk 1 above)
Comment (aesharpe) [reply src/pudl/helpers.py:1107]: yep!

**Comment by aesharpe** [comment]
Not sure why the tests are failing. Zane thinks it might be a problem with our integration with Codacy.

**Pull request merged**